frontend/src/components/pages/FridgeManagement/FridgeCheckIns/EditCheckin.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  Button,
+  Center,
+  CloseButton,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Spinner,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { format, isAfter, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { CreateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const EditCheckInPage = (): JSX.Element => {
+  const { id } = useParams<{ id: string }>();
+  const [date, setDate] = useState<string>("");
+  const [startTime, setStartTime] = useState<string>("");
+  const [endTime, setEndTime] = useState<string>("");
+  const [notes, setNotes] = useState<string | undefined>("");
+  const [timeRangeError, setTimeRangeError] = useState("");
+  const toast = useToast();
+  const history = useHistory();
+
+  const getCheckInData = async () => {
+    const checkInResponse = await CheckInAPIClient.getCheckInsById(id);
+    setDate(checkInResponse.startDate);
+    setStartTime(format(new Date(checkInResponse.startDate), "kk:mm"));
+    setEndTime(format(new Date(checkInResponse.endDate), "kk:mm"));
+    setNotes(checkInResponse.notes);
+  };
+
+  const validateForm = () => {
+    let newError = "";
+    let valid = true;
+    if (
+      isAfter(
+        parse(startTime, "kk:mm", new Date()),
+        parse(endTime, "kk:mm", new Date()),
+      )
```

Comment:
edge case: If we have it as the same time, then the backend doesnt update cuz startTime needs to be before endTime. We also need isEqual so that the error shows on this screen. Can you also add this to CreateCheckin.